### ISSUE TITLE
[Bug] No ros2 sensor topics are being created. I tried with Lidar and cemra. But, never was able to get lidar topics or camera topics. 

### ISSUE THREAD

**Issue by rahgirrafi** [PX4/PX4-Autopilot#23975]
### Describe the bug

I was trying to setup PX4 with ros2. I follower the ROS2 User Guide from PX4. 

I used MicroXRCEAgent to listen to ros2 topics.
```bash
MicroXRCEAgent udp4 -p 8888
```
Then I launch the x500_lidar_2d.
`
make px4_sitl gz_x500_lidar_2d
`

But I only get these ros2 topics. 
/fmu/in/actuator_motors
/fmu/in/actuator_servos
/fmu/in/arming_check_reply
/fmu/in/aux_global_position
/fmu/in/config_control_setpoints
/fmu/in/config_overrides_request
/fmu/in/goto_setpoint
/fmu/in/manual_control_input
/fmu/in/message_format_request
/fmu/in/mode_completed
/fmu/in/obstacle_distance
/fmu/in/offboard_control_mode
/fmu/in/onboard_computer_status
/fmu/in/register_ext_component_request
/fmu/in/sensor_optical_flow
/fmu/in/telemetry_status
/fmu/in/trajectory_setpoint
/fmu/in/unregister_ext_component
/fmu/in/vehicle_attitude_setpoint
/fmu/in/vehicle_command
/fmu/in/vehicle_command_mode_executor
/fmu/in/vehicle_mocap_odometry
/fmu/in/vehicle_rates_setpoint
/fmu/in/vehicle_thrust_setpoint
/fmu/in/vehicle_torque_setpoint
/fmu/in/vehicle_trajectory_bezier
/fmu/in/vehicle_trajectory_waypoint
/fmu/in/vehicle_visual_odometry
/fmu/out/battery_status
/fmu/out/estimator_status_flags
/fmu/out/failsafe_flags
/fmu/out/manual_control_setpoint
/fmu/out/position_setpoint_triplet
/fmu/out/sensor_combined
/fmu/out/timesync_status
/fmu/out/vehicle_attitude
/fmu/out/vehicle_control_mode
/fmu/out/vehicle_global_position
/fmu/out/vehicle_gps_position
/fmu/out/vehicle_land_detected
/fmu/out/vehicle_local_position
/fmu/out/vehicle_odometry
/fmu/out/vehicle_status
/parameter_events
/rosout

I don't get the lidar topics. It's the same if I try with camera. 

To sum up, no sensor topic is being created. 

### Screenshot / Media

![Screenshot from 2024-11-19 23-12-46](https://github.com/user-attachments/assets/071434fa-3649-47cc-9913-80ee3ad9a5fd)

### Vehicle type

None

**Comment by beniaminopozzan** [comment]
You are confusing the PX4-ROS interface with the Gazebo-ROS interface.
The lidar data (and the camera data) is not passed to PX4 as PX4 has no reason for using it. Therefore it is correct that there is no `/fmu/out/***` topic associated to it.

If you want such data avaiable in ROS 2 you need to use the `ros_gz_package` from https://github.com/gazebosim/ros_gz

Imagine working with an actual drone equipped with a lidar sensor. You don't expect the lidar to be directly connected to the FCU but rather you'll have the lidar connected to you companion computer. In your companion computer you'll have the right ROS 2 driver for the sensors which are completely unrelated to PX4
